src/main/java/org/jenkinsci/plugins/envinject/EnvInjectAction.java

```diff
@@ -46,6 +47,9 @@ public void overrideAll(Map<String, String> all) {
     }
 
     public String getIconFileName() {
+        if (!EnvInjectPlugin.canViewInjectedVars(build)) {
+            return null;
```

Comment:
This annotation is being inherited from the <code>Action::getIconFileName()</code>, hence there is no need to refer it again. BTW the <code>@Override</code> annotation would be useful to make it more clear